docs/pr_etiquette.md

```diff
@@ -0,0 +1,162 @@
+delayed. Remember that reviewing PRs is taking time from someone else's day.
+
+If you must submit a large PR, try to at least make someone else aware of this
+fact, and arrange for their time to review and get the PR merged. It's not fair
+to the team to dump large pieces of work on their laps without warning.
+
+If you can rebase up a large PR into multiple smaller PRs, then do so.
+
+
+Reviewing Pull Requests
+-----
+
+It's a reviewers responsibility to ensure:
+
+* Commit history is excellent
+* Good changes are propagated quickly
+* Code review is performed
+* They understand what is being changed, from the perspective of someone examining the code in the future.
+
+### Keep the flow going
+
+Pull Requests are the fundamental unit of how we progress change. If PRs are
+getting clogged up in the system, either unreviewed or unmanaged, they are
+preventing a piece of work from being completed.
+
+As PRs clog up in the system, merges become more difficult, as other features
+and fixes are applied to the same codebase. This in turn slows them down
+further, and often completely blocks progress on a given codebase.
+
+There is a balance between flow and ensuring the quality of our PRs. As a
+reviewer you should make a call as to whether a code quality issue is
+sufficient enough to block the PR whilst the code is improved. Possibly it is
+more prudent to simply flag that the code needs rework, and raise an issue.
+
+Any quality issue that will obviously result in a bug should be fixed.
+
+### We are all reviewers
+
+To make sure PRs flow through the system speedily, we must scale the PR review
+process. It is not sufficient (or fair!) to expect one or two people to review
+all PRs to our code. For starters, it creates a blocker every time those people
+are busy.
+
+Hopefully with the above guidelines, we can all start sharing the responsibility of being a reviewer.
+
+NB: With this in mind - if you are the first to comment on a PR, you are that
+PRs reviewer. If you feel that you can no longer be responsible for the
+subsequent merge or closure of the PR, then flag this up in the PR
+conversation, so someone else can take up the role.
+
+There's no reason why multiple people cannot comment on a PR and review it, and
+this is to be encouraged.
+
+
+### Don't add to the PR yourself.
+
+It's sometimes tempting to fix a bug in a PR yourself, or to rework a section
+to meet coding standards, or just to make a feature better fit your needs.
+
+If you do this, you are no longer the reviewer of the PR. You are a
+collaborator, and so should not merge the PR.
+
+It is of course possible to find a new reviewer, but generally change will be
+speedier if you require the original submitter to fix the code themselves.
+Alternatively, if the original PR is 'good enough', raise the changes you'd
+like to see as separate stories/issues, and rework in your own PR.
+
+### It is not the reviewers responsibility to test the code
```

Comment:
This probably holds up internally, but I'm not sure it does for prs submitted outside of our team. If a community member puts up a PR, I think we need to test it before merging.

Reply:
yeah we don't want to scare out contributors, you are right